proposals/bip-001.md

```diff
@@ -0,0 +1,107 @@
+# BIP Purpose and Guidelines
+
+| Field | Value |
+| --- | --- |
+| BIP | 1 |
+| Title | BIP Purpose and Guidelines |
+| Author | Alfredo Beaumont |
+| Status | Draft |
+| Created | 2017-05-26 |
+| Updated | 2017-05-26 |
+
+## Abstract
+
+BIP stands for Babelfish Improvement Proposal. A BIP is a design document which
+describes a change proposal to Babelfish. It's initially designed specifically
+to propose changes to [UAST](https://doc.bblf.sh/uast/specification.html)
+components regarding code categorization through Roles or related entities, but
+its scope may become broader in the future. It's informally defined.
```

Comment:
I agree that including full API of every component is just too much, but what about Babelfish protocols (server and driver protocols). BIPs should be more about specs rather than implementation. So for me, it would not make sense to include all Go APIs, but it would make sense to include full protocol specs. Basically, it's everything that ends up in a proto file.